incubator/vault/templates/deployment.yaml

```diff
@@ -22,21 +28,29 @@ spec:
         {{- if .Values.vault.dev }}
         command: ["vault", "server", "-dev", "-dev-listen-address", "[::]:8200"]
         {{- else }}
-        command: ["vault", "server", "-config", "/vault/config/config.json"]
+        command: ["sh", "-c"]
+        args:
+          - export VAULT_CLUSTER_ADDR="http://$POD_IP:8201"; vault server -config /vault/config/config.json
```

Comment:
I'm not sure, but shouldn't be possible to also set this var at env block?
```
- name: VAULT_CLUSTER_ADDR
  value: "http://$POD_IP:8201"
```

Reply:
by the way, i've also set `VAULT_API_ADDR`

Reply:
Edit the var name. `$(POD_IP)`

Reply:
I'm not sure if it will be expanded. Also, don't you mean `${POD_IP}`?

Reply:
It is with `(` and it does get expanded! I"ve tested! 
(I also found weird, but i found that at some doc design.

```
$ kubectl exec -it vault-vault-75b9887f95-rnpsj sh
/ # ip addr
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN qlen 1
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
3: eth0@if56: <BROADCAST,MULTICAST,UP,LOWER_UP,M-DOWN> mtu 1460 qdisc noqueue state UP 
    link/ether 0a:58:0a:3c:02:35 brd ff:ff:ff:ff:ff:ff
    inet 10.60.2.53/24 scope global eth0
       valid_lft forever preferred_lft forever
    inet6 fe80::d451:5dff:fe87:dcd/64 scope link 
       valid_lft forever preferred_lft forever
/ # echo $VAULT_CLUSTER_ADDR
http://10.60.2.53:8201
/ # %                                                                                                                                                                                                                                          (venv) 
$ kubectl describe pod vault-vault-75b9887f95-rnpsj  | grep VAULT_CLUSTER_ADDR
      VAULT_CLUSTER_ADDR:  http://$(POD_IP):8201
```

Reply:
Nice @sfrique, didn't know that worked. Updated.